node/node.go

```diff
@@ -334,21 +328,174 @@ func NewNode(
 
 		n.RelayClient.Store(relayClient)
 
-		blockNumber, err := tx.GetCurrentBlockNumber(context.Background())
+		blockNumber, err := tx.GetCurrentBlockNumber(ctx)
 		if err != nil {
 			return nil, fmt.Errorf("failed to get block number: %w", err)
 		}
-		quorumCount, err := tx.GetQuorumCount(context.Background(), blockNumber)
+		quorumCount, err := tx.GetQuorumCount(ctx, blockNumber)
 		if err != nil {
 			return nil, fmt.Errorf("failed to get quorum count: %w", err)
 		}
 		n.QuorumCount.Store(uint32(quorumCount))
+
+		n.BlobVersionParams.Store(blobVersionParams)
+	}
+
+	n.startPprof()
+	n.startMetrics()
+	n.startNodeAPI()
+	n.startV1()
+	n.startV2(ctx)
+
+	n.CurrentSocket = n.buildSocket()
+	if n.Config.RegisterNodeAtStart {
+		err = n.registerValidator(ctx, n.CurrentSocket)
+		if err != nil {
+			return nil, fmt.Errorf("failed to register validator: %w", err)
+		}
+	} else {
+		n.checkValidatorRegistration(ctx, n.CurrentSocket)
 	}
 
-	n.BlobVersionParams.Store(blobVersionParams)
+	n.startNodeIPUpdater(ctx)
+
 	return n, nil
 }
 
+// Start goroutines that periodically check and update the node's public IP address on-chain.
+func (n *Node) startNodeIPUpdater(ctx context.Context) {
+	// Start the Node IP updater only if the PUBLIC_IP_PROVIDER is greater than 0.
+	if n.Config.PubIPCheckInterval > 0 && n.Config.EnableV1 && n.Config.EnableV2 {
```

Comment:
Should this really be `n.Config.EnableV1 && n.Config.EnableV2`? Seems like `||` might be correct. This was preexisting, but might be a bug